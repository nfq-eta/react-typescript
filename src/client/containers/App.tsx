import * as React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import CheckBox, { IItem } from '../components/CheckBox';
import * as CheckboxItemsJson from '../components/tests/__mocks__/CheckboxItems.json';
import * as CheckboxSelectedItemsJson from '../components/tests/__mocks__/CheckboxSelectedItems.json';
import { addCheckbox } from '../modules/checkbox/actions';
import { IRootState } from '../core/reducers';

export interface IAppProps {
    actions: typeof addCheckbox;
    items: IItem[];
}

export interface IAppState {
    items: IItem[];
}

// @connect(mapStateToProps, mapDispatchToProps)
class App extends React.Component<IAppProps, IAppState> {
    selectedItems = new Map(CheckboxSelectedItemsJson);

    constructor() {
        super();

        this.state = { items: CheckboxItemsJson };
    }

    handleClick = (item: IItem, checked: boolean) => {
        if (checked) {
            this.selectedItems.set(item.value, item);
        } else {
            this.selectedItems.delete(item.value);
        }
    };

    isSelected(item: IItem): boolean {
        return this.selectedItems.has(item.value);
    }

    handleAdd = () => {
        this.props.actions({
            value: 'demo',
            label: 'Demo',
        });
    };

    render() {
        return (
            <div>
                <h1>App</h1>
                <button onClick={this.handleAdd}>Add more</button>
                {this.props.items.map(item => (
                    <CheckBox
                        item={item}
                        key={'a' + item.value}
                        checked={this.isSelected(item) || false}
                        handleClick={this.handleClick}
                    />
                    ))
                }
            </div>
        );
    }
}

function mapStateToProps(state: IRootState) {
    return ({
        items: state.checkBoxItems,
    });
}

function mapDispatchToProps(dispatch: any) {
    return ({
        actions: bindActionCreators(addCheckbox as any, dispatch),
    });
}

export default connect(mapStateToProps, mapDispatchToProps)(App);

export { App as AppDisconnected };
