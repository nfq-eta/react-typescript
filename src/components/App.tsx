import * as React from 'react';
import CheckBox, { IItem } from './CheckBox';

import CheckboxItemsJson = require('./tests/__mocks__/CheckboxItems.json');
import CheckboxSelectedItemsJson = require('./tests/__mocks__/CheckboxSelectedItems.json');

export interface IAppProps {
    title?: string;
}

export interface IAppState {
    items: IItem[];
}

export default class App extends React.Component<IAppProps, IAppState> {
    selectedItems = new Map(CheckboxSelectedItemsJson);

    constructor() {
        super();

        this.state = { items: CheckboxItemsJson };
    }

    handleClick = (item: IItem, checked: boolean) => {
        if (checked) {
            this.selectedItems.set(item.id, item);
        } else {
            this.selectedItems.delete(item.id);
        }
    };

    isSelected(item: IItem): boolean {
        return this.selectedItems.has(item.id);
    }

    render() {
        return (
            <div>
                <h1>{this.props.title || ''}</h1>
                {this.state.items.map(item => (
                    <CheckBox
                        item={item}
                        key={item.id}
                        checked={this.isSelected(item) || false}
                        handleClick={this.handleClick}
                    />
                    ))
                }
            </div>
        );
    }
}
