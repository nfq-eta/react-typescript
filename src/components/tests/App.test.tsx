import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';
import {} from 'jest';

import App from '../App';

describe('App.tsx', () => {
    it('renders without crashing', () => {
        shallow(<App title="Test" />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('Title should be visible', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(wrapper.find('h1').text()).toContain('Test');
    });

    it('Title should not be visible', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.find('h1').text().length).toBe(0);
    });

    it('should handle click', () => {
        const wrapper = shallow(<App />);
        const instance = wrapper.instance() as App;
        const params = { id: 1, label: 'Demo' };

        instance.handleClick(params, true);
        expect(instance.selectedItems.has(1)).toBeTruthy();

        instance.handleClick(params, false);
        expect(instance.selectedItems.has(1)).toBeFalsy();
    });
});
