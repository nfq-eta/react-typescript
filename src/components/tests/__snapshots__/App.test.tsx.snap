// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js matches snapshot 1`] = `
<div>
  <h1>
    Test
  </h1>
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 1,
        "label": "Foo",
      }
    }
    key="1"
  />
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 2,
        "label": "Buzz",
      }
    }
    key="2"
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 3,
        "label": "Fuzz",
      }
    }
    key="3"
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 4,
        "label": "Fizz",
      }
    }
    key="4"
  />
  <CheckBox
    checked={false}
    handleClick={[Function]}
    item={
      Object {
        "id": 5,
        "label": "FizzBuzz",
      }
    }
    key="5"
  />
  <CheckBox
    checked={true}
    handleClick={[Function]}
    item={
      Object {
        "id": 6,
        "label": "Boo",
      }
    }
    key="6"
  />
</div>
`;
